src/Unused/CLI/Util.hs

```diff
@@ -1,11 +1,28 @@
 module Unused.CLI.Util
     ( resetScreen
+    , withInterruptHandler
     , module System.Console.ANSI
     ) where
 
 import System.Console.ANSI
+import Control.Exception (throwTo)
+import System.Posix.Signals (Handler(Catch), installHandler, keyboardSignal)
+import Control.Concurrent (ThreadId, myThreadId)
+import System.Exit (ExitCode(ExitFailure))
 
 resetScreen :: IO ()
 resetScreen = do
     clearScreen
     setCursorPosition 0 0
+
+withInterruptHandler :: IO () -> IO ()
+withInterruptHandler body = do
+    tid <- myThreadId
+    _ <- installHandler keyboardSignal (Catch (handleInterrupt tid)) Nothing
```

Comment:
I have `-Wall -Werror` turned on (as I just found out, it's only in exe, and not in library, which was driving me nuts because I was seeing them when running `stack ghci` but not `stack install`!)